### PR TITLE
Detaching from events dependency, switching to using EventTarget

### PR DESCRIPTION
**Pull request by MaksimLavrenyuk**

### CODE REVIEW

**Review by jessetane — CHANGES_REQUESTED** [review]
looking good! only thought is that the detail property for error should literally be `error` instead of `err`

File: index.js [hunk 1]
```diff
@@ -109,12 +120,12 @@ export default class Queue extends EventEmitter {
         }
 
         if (err) {
-          this.emit('error', err, job);
+          this.dispatchEvent(new QueueEvent('error', { err, job }))
```
[the remark this answers is not in the document]
Comment (MaksimLavrenyuk) [reply index.js]: Yeah, sure.

**Comment by jessetane** [comment]
thanks. so this is where i would use `git rebase -i` to squash these new commits with their earlier counterparts so the history stays clean. not trying to be annoying but i think most open source projects will expect this in the PR feedback / iteration process. if it's too much hassle though i can live with it as is

**Comment by jessetane** [comment]
haven't read this but it came up with a quick search, maybe helpful: https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history#interactive-rebase

**Comment by MaksimLavrenyuk** [comment]
Thanks for the tip! Yeah, I'll do it https://github.com/jessetane/queue/pull/92#issuecomment-1428520093, it's time for me to figure it out.

**Comment by jessetane** [comment]
note that when you alter history you'll have to force push to your branch, but this is normal

**Comment by MaksimLavrenyuk** [comment]
@jessetane

Check this

**Comment by jessetane** [comment]
what's next?

**Comment by MaksimLavrenyuk** [comment]
We need to switch to Promise instead of callback. After that I would like to add a d.ts file to support projects that use typescript.

**Comment by jessetane** [comment]
sounds good, let's tackle those in separate PR's. ideally we can continue to support promises _and_ callbacks, but maybe take into account concerns raised in #85 and #34?

**Comment by MaksimLavrenyuk** [comment]
@jessetane 

I apologize for the delay, my main work takes a lot of time.

For issue https://github.com/jessetane/queue/issues/34, I created a [PR](https://github.com/jessetane/queue/pull/93).
For issue https://github.com/jessetane/queue/issues/85, there are questions that need to be discussed. I will write to the issue itself.

**Pull request merged**